File: C#/QuickSort.cs
using System;
using System.Diagnostics;

class Program
{
    // Partition function (Lomuto scheme)
    static int Partition(int[] arr, int low, int high)
    {
        int pivot = arr[high];
        int i = low - 1;

        for (int j = low; j < high; j++)
        {
            if (arr[j] <= pivot)
            {
                i++;
                int temp = arr[i];
                arr[i] = arr[j];
                arr[j] = temp;
            }
        }

        int temp2 = arr[i + 1];
        arr[i + 1] = arr[high];
        arr[high] = temp2;

        return i + 1;
    }

    // Quick Sort recursive function
    static void QuickSort(int[] arr, int low, int high)
    {
        if (low < high)
        {
            int pi = Partition(arr, low, high);
            QuickSort(arr, low, pi - 1);
            QuickSort(arr, pi + 1, high);
        }
    }

    static void Main()
    {
        Console.Write("Enter number of elements: ");
        int n = int.Parse(Console.ReadLine());

        int[] arr = new int[n];

        if (n > 0)
        {
            Console.WriteLine("Enter elements separated by space:");
            string inputLine = Console.ReadLine();
            string[] parts = inputLine.Split(new char[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);

            for (int i = 0; i < n; i++)
                arr[i] = int.Parse(parts[i]);
        }

        Stopwatch sw = Stopwatch.StartNew();
        if (n > 0)
            QuickSort(arr, 0, n - 1);
        sw.Stop();

        Console.WriteLine("\nSorted array:");
        for (int i = 0; i < n; i++)
            Console.Write(arr[i] + " ");
        Console.WriteLine();

        Console.WriteLine("\nExecution time: " + sw.Elapsed.TotalSeconds + " seconds");
    }
}

File: Rust/QuickSort.rs
use std::io::{self, Read};
use std::time::Instant;

// Partition function 
fn partition(arr: &mut [i32], low: isize, high: isize) -> isize {
    let pivot = arr[high as usize];
    let mut i = low - 1;

    for j in low..high {
        if arr[j as usize] <= pivot {
            i += 1;
            arr.swap(i as usize, j as usize);
        }
    }

    arr.swap((i + 1) as usize, high as usize);
    i + 1
}

// Quick Sort recursive function
fn quick_sort(arr: &mut [i32], low: isize, high: isize) {
    if low < high {
        let pi = partition(arr, low, high);
        quick_sort(arr, low, pi - 1);
        quick_sort(arr, pi + 1, high);
    }
}

fn main() {
    // Read all input at once
    let mut input = String::new();
    io::stdin().read_to_string(&mut input).unwrap();
    let mut iter = input.split_whitespace();

    println!("Enter number of elements:");
    let n: usize = match iter.next() {
        Some(v) => v.parse().unwrap(),
        None => return,
    };

    println!("Enter elements separated by space:");
    let mut arr: Vec<i32> = Vec::with_capacity(n);
    for _ in 0..n {
        if let Some(v) = iter.next() {
            arr.push(v.parse().unwrap());
        }
    }

    let start = Instant::now();
    if n > 0 {
        let len = arr.len();
        quick_sort(&mut arr, 0, (len - 1) as isize);
    }
    let duration = start.elapsed();

    println!("\nSorted array:");
    for x in &arr {
        print!("{} ", x);
    }
    println!();

    println!(
        "\nExecution time: {:.9} seconds",
        duration.as_secs_f64()
    );
}
